src/main/java/net/glease/ggfab/mte/MTE_AdvAssLine.java

```diff
@@ -766,8 +783,33 @@ public CheckRecipeResult checkProcessing() {
                         mMaxProgresstime = laserOverclock.getDuration();
                     }
                 }
+                // Save this for batch mode parallel calculations
+                int timePerSlice = mMaxProgresstime;
                 // correct the recipe duration
                 mMaxProgresstime *= recipe.mInputs.length;
+
+                // Finally apply batch mode parallels if possible.
+                // For this we need to verify the first item slot and all fluids slots have enough resources
+                // to execute parallels.
+                // Note that we skip this entirely if the time for each slice is more than 128 ticks,
+                // since in this case the amount of batches will always be 1
+                if (super.isBatchModeEnabled() && timePerSlice < BATCH_MODE_MIN_TICK_TIME) {
+                    // Calculate parallel based on time per slice, and the amount of items in the first slot.
+                    // If there is not enough fluid, no batching will be done.
+
+                    ItemStack firstItemSlot = getInputBusContent(0);
+                    int recipesAvailable = Math.floorDiv(firstItemSlot.stackSize, recipe.mInputs[0].stackSize);
+                    int desiredBatches = (int) Math.ceil((float) BATCH_MODE_MIN_TICK_TIME / (float) timePerSlice);
+                    int parallel = Math.min(recipesAvailable, desiredBatches);
+                    // We no longer need to check if we have enough items in the first slot, as this is
+                    // guaranteed by taking the minimum earlier.
+                    if (hasAllFluids(recipe, parallel)) {
+                        this.currentRecipeParallel = parallel;
```

Comment:
this method is called in onRunningTick() as well (when first slice is idle). doing parallel here can have very bad outcome.